src/btree/bt_curnext.c

```diff
@@ -101,7 +102,14 @@ __cursor_fix_next(WT_CURSOR_BTREE *cbt, bool newpage)
 	    cbt->ins_head, cbt->ins_stack, cbt->next_stack, cbt->recno);
 	if (cbt->ins != NULL && cbt->recno != WT_INSERT_RECNO(cbt->ins))
 		cbt->ins = NULL;
-	upd = cbt->ins == NULL ? NULL : __wt_txn_read(session, cbt->ins->upd);
+
+	upd = NULL;			/* -Werror=maybe-uninitialized */
+	if (cbt->ins != NULL) {		/* -Werror=dangling-else */
+		if (__wt_txn_read(session, cbt->ins->upd, &upd) ==
+		    WT_VISIBLE_PREPARE)
+			return (WT_PREPARE_CONFLICT);
+	} else
+		upd = NULL;
```

Comment:
Since we are already initialising upd to NULL, we don't need else statement here.
Same comment for the other instances of this below.